core/src/main/scala/org/apache/spark/deploy/Client.scala

```diff
@@ -190,13 +190,15 @@ private class ClientEndpoint(
                 logDebug(s"State of driver $submittedDriverID is ${state.get}, " +
                   s"continue monitoring driver status.")
               }
-            }
-        }
-      } else {
+          }
+      }
+    } else if (!exception.isEmpty && Utils.responseFromBackup(exception.get.getMessage())) {
```

Comment:
We could simplify this a bit:
```suggestion
    } else if (exception.exists(e => Utils.responseFromBackup(e.getMessage))) {
```

Reply:
Yes, I agree with it.